src/staking-router-modules/community-module.service.ts

```diff
@@ -0,0 +1,134 @@
+import { Inject, Injectable } from '@nestjs/common';
+import {
+  RegistryKeyStorageService,
+  RegistryKey,
+  RegistryOperator,
+  RegistryOperatorStorageService,
+} from '../common/registry';
+
+import { CSMKeyRegistryService } from '../common/registry-csm';
+import { LOGGER_PROVIDER, LoggerService } from '../common/logger';
+import { QueryOrder } from '@mikro-orm/core';
+import { StakingModuleInterface } from './interfaces/staking-module.interface';
+import { KeysFilter, OperatorsFilter } from './interfaces/filters';
+
+@Injectable()
+export class CommunityModuleService implements StakingModuleInterface {
+  constructor(
+    @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
+    protected readonly keyRegistryService: CSMKeyRegistryService,
+    protected readonly keyStorageService: RegistryKeyStorageService,
+    protected readonly operatorStorageService: RegistryOperatorStorageService,
+  ) {}
+
+  public async update(moduleAddress: string, blockHash: string): Promise<void> {
+    await this.keyRegistryService.update(moduleAddress, blockHash);
+  }
+
+  public async operatorsWereChanged(
+    moduleAddress: string,
+    fromBlockNumber: number,
+    toBlockNumber: number,
+  ): Promise<boolean> {
+    return await this.keyRegistryService.operatorsWereChanged(moduleAddress, fromBlockNumber, toBlockNumber);
+  }
+
+  public async updateOperators(moduleAddress: string, blockHash: string): Promise<void> {
+    await this.keyRegistryService.updateOperators(moduleAddress, blockHash);
+  }
+
+  public async getCurrentNonce(moduleAddress: string, blockHash: string): Promise<number> {
+    const nonce = await this.keyRegistryService.getStakingModuleNonce(moduleAddress, blockHash);
```

Comment:
return await this.keyRegistryService.getStakingModuleNonce(moduleAddress, blockHash);